grc/core/blocks/block.py

```diff
@@ -539,6 +540,30 @@ def is_virtual_sink(self):
     def is_virtual_source(self):
         return self.key == 'virtual_source'
 
+    @property
+    def deprecated(self):
```

Comment:
Yeah, I was torn for that exact reason. We can convert that into an actual getter later on, if there's multiple sources of truth for that e.g. if we should come around to implementing something like block servers. Will change this to be plain.

Reply:
do we know all categories a block is in  in _build.py? Can parsing another block_tree lead to reorganization later on?

Reply:
good point, categories may be overwritten after the block class was created. Once, directly after, see https://github.com/gnuradio/gnuradio/blob/0b3dc1b157b709fcd64a80f13ec604bda3f86e66/grc/core/platform.py#L169

one could make sure block_tree will get loaded first and pass the final category directly to the class builder. I doubt that worth the effort. `__init__` then.

It occurs to me, that you might want to support deprecation as a block flag. pull that into a class attribute `is_deprecated`, an have the library loader set that when it overwrites the category.

Reply:
> It occurs to me, that you might want to support deprecation as a block flag. pull that into a class attribute is_deprecated, an have the library loader set that when it overwrites the category.

Creating a deprecated block flag would be helpful to UI design as it could be leveraged in design. Instead of deprecation being communicated via block colour/block stroke/something arbitary, it could be communicated by that flag.

![image](https://user-images.githubusercontent.com/1899506/115447064-f6937480-a20f-11eb-94b5-d6561655f4d8.png)
In this instance the block is deprecated, but also by-passed.

This deprecated flag would be set, until the the block was no longer deprecated, right?

Reply:
...not talking about UI here. This discussion is about how to efficiently handle this in the backend ("grc/core"). 
How this information is used in the GUI part of GRC is another matter....

Reply:
Right, but that flag would be available to be used in the GUI, right?

Reply:
Of course. 

Note, with the term "flag" I mean attribute of block descriptions (YAML) as will as a corresponding class attribute in the backend block classes. User visible flags (on blocks) would be a different concept.

Reply:
> It occurs to me, that you might want to support deprecation as a block flag. pull that into a class attribute `is_deprecated`, an have the library loader set that when it overwrites the category.

Exactly (I think I've even hinted at that :) )! Generally, I'd rather have a consistent convention on flags (like deprecated, rate_limiting, …) than have to infer it from category names.

> `__init__` then.

Ah, that's great. You saying that implies at ctor time the categories are fixed!